### PR TITLE
backport: trivial 2024 06 11

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Trivial backports

## How Has This Been Tested?
Built and ran tests locally; p2p_addr_relay.py fails locally. Not sure why

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Comment by knst** [comment]
> Built and ran tests locally; p2p_addr_relay.py fails locally. Not sure why

succeed locally, CI also succeed
```
$ test/functional/p2p_addr_relay.py 
2024-06-12T18:07:56.658000Z TestFramework (INFO): Initializing test directory /tmp/dash_func_test_ojh8qt70
2024-06-12T18:07:58.553000Z TestFramework (INFO): Send an addr message that is too large
2024-06-12T18:08:00.058000Z TestFramework (INFO): Test address relay
2024-06-12T18:08:00.058000Z TestFramework (INFO): Check that addr message content is relayed and added to addrman
2024-06-12T18:08:12.570000Z TestFramework (INFO): Check relay of addresses received from outbound peers
2024-06-12T18:08:15.572000Z TestFramework (INFO): Check that the first addr message received from an outbound peer is not relayed
2024-06-12T18:08:16.073000Z TestFramework (INFO): Check that subsequent addr messages sent from an outbound peer are relayed
2024-06-12T18:08:17.074000Z TestFramework (INFO): Check address relay to outbound peers
2024-06-12T18:08:19.576000Z TestFramework (INFO): Check that addresses are relayed to full outbound peers
2024-06-12T18:08:19.576000Z TestFramework (INFO): Check that addresses are not relayed to block-relay-only outbound peers
2024-06-12T18:08:20.079000Z TestFramework (INFO): Check that we only relay addresses to inbound peers who have previously sent us addr related messages
2024-06-12T18:08:24.586000Z TestFramework (INFO): After blackhole peer sends addr message, it becomes eligible for addr gossip
2024-06-12T18:08:30.598000Z TestFramework (INFO): Test getaddr behavior
2024-06-12T18:08:30.598000Z TestFramework (INFO): Check that we send a getaddr message upon connecting to an outbound-full-relay peer
2024-06-12T18:08:32.098000Z TestFramework (INFO): Check that we do not send a getaddr message upon connecting to a block-relay-only peer
2024-06-12T18:08:33.599000Z TestFramework (INFO): Check that we answer getaddr messages only from inbound peers
2024-06-12T18:08:37.341000Z TestFramework (INFO): Test addr relay in -blocksonly mode
2024-06-12T18:08:42.347000Z TestFramework (INFO): Check that we send getaddr messages
2024-06-12T18:08:43.847000Z TestFramework (INFO): Check that we relay address messages
2024-06-12T18:08:48.855000Z TestFramework (INFO): Test rate limiting of addr processing for outbound-full-relay peers
2024-06-12T18:08:52.369000Z TestFramework (INFO): Test rate limiting of addr processing for block-relay-only peers
2024-06-12T18:08:56.394000Z TestFramework (INFO): Test rate limiting of addr processing for inbound peers
2024-06-12T18:09:00.409000Z TestFramework (INFO): Stopping nodes
2024-06-12T18:09:01.410000Z TestFramework (INFO): Cleaning up /tmp/dash_func_test_ojh8qt70 on exit
2024-06-12T18:09:01.410000Z TestFramework (INFO): Tests successful
```

**Review by knst — APPROVED** [review]
utACK cac358975b0fc3443539fc4186b5888092106bb3

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/test_framework/test_shell.py [hunk 1]
```diff
@@ -16,6 +16,9 @@ class TestShell:
     start a single TestShell at a time."""
 
     class __TestShell(BitcoinTestFramework):
+        def add_options(self, parser):
```
[the remark this answers is not in the document]
Comment (UdjinM6) [reply test/functional/test_framework/test_shell.py:19]: `add_wallet_options` was introduced in the commit in question so you call a non-existent function here 🤷‍♂️

**Review by knst — APPROVED** [review]
utACK fb8a4db8f67966a157e24ce96144127200b4a7ee

**Pull request merged**